lib/matplotlib/tests/test_axes.py

```diff
@@ -622,6 +624,16 @@ def test_annotate_default_arrow():
     assert ann.arrow_patch is not None
 
 
+def test_annotate_signature():
+    """Check that the signature of Axes.annotate() matches Annotation."""
+    fig, ax = plt.subplots()
+    annotate_params = inspect.signature(ax.annotate).parameters
```

Comment:
Can you not inspect `Axes.annotate` like is done for `Annotation`?

Reply:
Yes, I could. but the comparison is more difficult because `Axes.annotate` is an unbound method and thus has an extra `self` parameter, which makes the comparison a bit more difficult.